pkg/resourceprovider/provider.go

```diff
@@ -398,15 +413,41 @@ func (p *resourceProvider) queryBoth(now pmodel.Time, resource schema.GroupResou
 	wg.Wait()
 
 	if cpuErr != nil {
-		return nsQueryResults{
-			namespace: namespace,
-			err:       fmt.Errorf("unable to fetch node CPU metrics: %v", cpuErr),
+		if promErr, ok := cpuErr.(*client.Error); ok {
+			return nsQueryResults{
+				namespace: namespace,
+				err:      &client.Error{
+					Type: promErr.Type,
+					ErrorMsg:  fmt.Sprintf("unable to fetch node CPU metrics: %s", promErr.ErrorMsg),
+					Query: promErr.Query,
+					StatusCode: promErr.StatusCode,
+				},
+			}
+		} else {
+			// Generic error handling for other types of errors
+			return nsQueryResults{
+				namespace: namespace,
+				err:       fmt.Errorf("unable to fetch node CPU metrics: %v", cpuErr),
```

Comment:
this is not necessarily `node`-related right? It can also be pod? again, don't think it's necessary to distinguish node vs. pod here because the caller will be getnodemetrics or getpodmetrics, and we should log that info from the caller